utils/fs-operations.js

```diff
@@ -0,0 +1,84 @@
+const fs = require('fs');
+
+/**
+ * @summary Validate the DB directory path and throw an error if invalid
+ * @function
+ * @throws Throws an error if the path is not valid
+ * @param dbDirectoryPath
+ */
+const validateDBPath = (dbDirectoryPath) => {
+  if (!fs.existsSync(dbDirectoryPath)) {
+    throw new Error(`DB directory path: '${dbDirectoryPath}' is invalid`);
+  }
+};
+
+/**
+ * @summary Read a JSON file and return the contents as an object
+ * @function
+ * @param {string} filePath
+ * @returns {Object}
+ */
+const readJSONFile = (filePath) => {
+  if (fs.existsSync(filePath)) {
+    return JSON.parse(fs.readFileSync(filePath, 'utf8'));
+  }
+  return null;
+};
+
+/**
+ * @summary Write an object to a JSON file with formatting
+ * @function
+ * @param {string} filePath
+ * @param {Object} data
+ * @param {Object} options
+ */
+const writeJSONFile = (filePath, data, options = {}) => {
+  fs.writeFileSync(filePath, JSON.stringify(data, null, 2), options);
+};
+
+/**
+ * @summary Initialize a student directory with a counter starting at '1'
+ * @function
+ * @param {string} studentDirPath
+ * @param {string} counterFilePath
+ */
+const initStudentDir = (studentDirPath, counterFilePath) => {
+  if (!fs.existsSync(studentDirPath)) {
+    fs.mkdirSync(studentDirPath);
+    fs.writeFileSync(counterFilePath, '1\n', { flag: 'wx' });
+  }
+};
+
+/**
+ * @summary Get the value of a counter as a string
+ * @function
+ * @throws Throws an error if the counter file is invalid
+ * @param {string} counterFilePath
+ * @returns {string}
+ */
+const getCounter = (counterFilePath) => {
+  // contents of counter file should be only digits followed by a newline
+  const counterRegExp = /^(\d+)(\n)$/;
+  const contents = fs.readFileSync(counterFilePath).toString();
+  const match = counterRegExp.exec(contents);
+  if (!match) {
```

Comment:
I don't think we need to in this situation. Since `match` will be `null` if the regular expression doesn't match, and `match[1]` must have some value if a match is successful, I don't think there could be a situation where `match[1]` would be `undefined` or `null` while `match` is not `null`.